src/celeritas/phys/RootEventSampler.cc

```diff
@@ -0,0 +1,65 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/phys/RootEventSampler.cc
+//---------------------------------------------------------------------------//
+#include "RootEventSampler.hh"
+
+#include <algorithm>
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct \c RootEventReader , and initialize sampling conditions.
+ */
+RootEventSampler::RootEventSampler(std::string const& filename,
+                                   SPConstParticles params,
```

Comment:
```suggestion
                                   SPConstParticles particles,
```
(and in usage below) to avoid ambiguity